src/painter/canvas.rs

```diff
@@ -236,9 +233,11 @@ impl<'w, 's> CanvasCommands<'w> for Commands<'w, 's> {
             config.sampler.clone(),
             config.hdr,
         );
-        (
-            handle.clone(),
-            self.spawn(CanvasBundle::new(handle, config)),
-        )
+        let hdr = config.hdr;
+        let mut entity = self.spawn(CanvasBundle::new(handle.clone(), config));
+        if hdr {
+            entity.insert(Hdr);
+        }
+        (handle, entity)
```

Comment:
This works around `Hdr` being a component, but there's still a footgun if you call `CanvasBundle::new` manually as it won't respect the `config.hdr` setting.

I don't know how to make it foolproof because of a bunch of bevy limitations:
* Bundles can't have optional components (https://github.com/bevyengine/bevy/issues/2157)
* Required components can't get data from the world, so it seems like they can't be used because of the render target image

The only solution I see is to document the behavior, but I might have missed a better solution.